nodeadm/internal/kubelet/config.go

```diff
@@ -249,9 +257,19 @@ func (ksc *kubeletConfig) withCloudProvider(cfg *api.NodeConfig, flags map[strin
 
 // When the DefaultReservedResources flag is enabled, override the kubelet
 // config with reserved cgroup values on behalf of the user
-func (ksc *kubeletConfig) withDefaultReservedResources() {
+func (ksc *kubeletConfig) withDefaultReservedResources(cfg *api.NodeConfig) {
 	ksc.SystemReservedCgroup = ptr.String("/system")
 	ksc.KubeReservedCgroup = ptr.String("/runtime")
+	maxPods, ok := MaxPodsPerInstanceType[cfg.Status.Instance.Type]
+	if !ok {
+		return
+	}
+	ksc.MaxPods = int32(maxPods)
+	ksc.KubeReserved = map[string]string{
```

Comment:
Ideally we should have some fall back for the `kubeReserved` fields in rare cases we don't have `maxPods`. 

I don't feel like rushing to add the dynamic max pods calculation (and adding an ec2 call) is the best thing, but we could do it. 

Maybe we use the default `110` for max pods in the memory-to-reserve calc, but not set it in the kubelet config for now. This is just a rare edge case but i want to have a sane default 🙃

Reply:
I think this rare edge case would happen when new instance type launched and our AMI does not have the latest max pod.
Make sure I understand this:
- Do you mean instead of default both `MaxPods` and the `kubeReserved.memory`, we just use `110` for kubeReserved.memory calc when the instance type is missing?
- Without setting MaxPods, what would be the upper limit of it?
- Where this 110 value comes from?

As I think if we want to add default value, we should default both of them

Reply:
i think the case only gets hit when an old AMI is used on a new instance type which isn't in the version's packaged `eni-max-pods`. just making sure we don't completely give up in that scenario

Reply:
> I don't feel like rushing to add the dynamic max pods calculation (and adding an ec2 call) is the best thing, but we could do it.

I don't see much benefit in removing this fallback, and I think it'd be a surprising (and annoying) difference between AL2 and AL2023. It's easy for us to implement. If someone doesn't want to give their node role the `ec2:DescribeInstanceTypes` permission, we'll get an error and *then* we can fallback to `110`.

Reply:
if we *are* going to implement it, should we just be generating a file with all the instance descriptors like `EniLimit`, `CPUCount`, etc , similar to the [vpc_limits file](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/vpc/vpc_ip_resource_limit.go)?

Reply:
I think the eni-max-pods would be enough for most of the cases and don't  really need to replicate instance descriptors for all types. And inflight calculation for new types should be enough and the eni-max-pods should get updated in next AMI